src/tiled/propertybrowser.cpp

```diff
@@ -551,26 +551,30 @@ void PropertyBrowser::propertiesChanged(Object *object)
         updateCustomProperties();
 }
 
-void PropertyBrowser::componentAdded(Object *, const QString &)
+void PropertyBrowser::componentAdded()
 {
     addComponents();
 }
 
-void PropertyBrowser::removeComponent(Object *object, const QString &componentName)
+void PropertyBrowser::componentRemoved(const QList<Object *> &, const QString &name)
 {
-    if (object != mObject)
-        return;
+    if (mObject && !mObject->hasComponent(name)) {
+        // remove and delete component properties
+        const auto componentProperties = mMapComponentPropertyField.take(name);
+        for (QtProperty *property : componentProperties) {
+            mMapComponentProperty.remove(property);
+            mPropertyToId.remove(property);
+            delete property;
+        }
 
-    // remove and delete component properties
-    const auto componentProperties = mMapComponentPropertyField.take(componentName);
-    for (QtProperty *property : componentProperties) {
-        mMapComponentProperty.remove(property);
-        mPropertyToId.remove(property);
-        delete property;
-    }
+        // remove and delete component group property
+        delete mComponents.take(name);
 
-    // remove and delete component group property
-    delete mComponents.take(componentName);
+    } else if (!Object::commonComponents(mDocument->currentObjects()).contains(name)) {
+        // object on current but no longer common, then it needs to be disabled
+        if (auto componentProperty = mComponents.value(name))
+            componentProperty->setEnabled(false);
```

Comment:
I haven't found a scenario where this would execute since I think the menu will have the item disabled. Anyways, good to have it.

Reply:
This is relevant when the action is triggered via undo/redo, which can happen when a different set of objects is selected.